components/net/filemanager_thread.rs

```diff
@@ -279,21 +218,34 @@ impl FileManagerStore {
         }
     }
 
-    fn select_file<UI>(&self,
-                       patterns: Vec<FilterPattern>,
-                       sender: IpcSender<FileManagerResult<SelectedFile>>,
-                       origin: FileOrigin,
-                       opt_test_path: Option<String>,
-                       ui: &UI)
-        where UI: UIProvider,
-    {
+    fn get_selected_files(&self,
+                            patterns: Vec<FilterPattern>,
+                            multiple_files: bool,
+                            embedder_proxy: EmbedderProxy) -> Option<Vec<String>> {
+        if opts::get().headless {
+            return None;
+        }
```

Comment:
Can you let the embedder take the decision to return None when we run in headless mode?

Reply:
Guess so, will try for it.